sympytorch/sympy_module.py

```diff
@@ -66,25 +66,29 @@ def fn_(*args):
 class _Node(torch.nn.Module):
     def __init__(self, *, expr, _memodict, _func_lookup, **kwargs):
         super().__init__(**kwargs)
-
+        
         self._sympy_func = expr.func
-
         if issubclass(expr.func, sympy.Float):
             self._value = torch.nn.Parameter(torch.tensor(float(expr)))
             self._torch_func = lambda: self._value
             self._args = ()
-        elif issubclass(expr.func, sympy.UnevaluatedExpr):
-            if len(expr.args) != 1 or not issubclass(expr.args[0].func, sympy.Float):
-                raise ValueError("UnevaluatedExpr should only be used to wrap floats.")
-            self.register_buffer('_value', torch.tensor(float(expr.args[0])))
-            self._torch_func = lambda: self._value
+        elif issubclass(expr.func, sympy.Rational):
+            self.register_buffer('_numerator', torch.tensor(expr.p, dtype=torch.float64))
+            self.register_buffer('_denominator', torch.tensor(expr.q, dtype=torch.float64))
```

Comment:
These should probably have an integer dtype. That'll be the default inferred anyway, so I think just `torch.tensor(expr.numerator)`, `torch.tensor(expr.denominator)` should suffice.

Reply:
problem with this is that by default, division of integer tensors will result in torch.float32 tensors, whereas SR's representation is usually 64 bits, thus resulting into precision errors.

Reply:
Even so, fixing float64 here will necessarilly "infect" the rest of the computation with float64, which may be undesirable.

Perhaps instead `torch.tensor(expr.numerator).float()`. This will then result in it being convertible to float64, if subsequently desired, via `model.to(torch.float64)`.